src/SarifCli/Rules/SarifValidationSkimmerBase.cs

```diff
@@ -0,0 +1,331 @@
+            {
+                Notification[] toolNotifications = run.ToolNotifications.ToArray();
+                string toolNotificationsPointer = runPointer.AtProperty(SarifPropertyName.ToolNotifications);
+
+                for (int iToolNotification = 0; iToolNotification < toolNotifications.Length; ++iToolNotification)
+                {
+                    Notification toolNotification = toolNotifications[iToolNotification];
+                    string toolNotificationPointer = toolNotificationsPointer.AtIndex(iToolNotification);
+
+                    Visit(toolNotification, toolNotificationPointer);
+                }
+            }
+
+            if (run.ConfigurationNotifications != null)
+            {
+                Notification[] configurationNotifications = run.ConfigurationNotifications.ToArray();
+                string configurationNotificationsPointer = runPointer.AtProperty(SarifPropertyName.ConfigurationNotifications);
+
+                for (int iConfigurationNotification = 0; iConfigurationNotification < configurationNotifications.Length; ++iConfigurationNotification)
+                {
+                    Notification configurationNotification = configurationNotifications[iConfigurationNotification];
+                    string configurationNotificationPointer = configurationNotificationsPointer.AtIndex(iConfigurationNotification);
+
+                    Visit(configurationNotification, configurationNotificationPointer);
+                }
+            }
+        }
+
+        private void Visit(Stack stack, string stackPointer)
+        {
+            if (stack.Frames != null)
+            {
+                StackFrame[] frames = stack.Frames.ToArray();
+                string framesPointer = stackPointer.AtProperty(SarifPropertyName.Frames);
+
+                for (int iFrame = 0; iFrame < frames.Length; ++iFrame)
+                {
+                    StackFrame frame = frames[iFrame];
+                    string framePointer = framesPointer.AtIndex(iFrame);
+
+                    Visit(frame, framePointer);
+                }
+            }
+        }
+
+        private void Visit(StackFrame frame, string framePointer)
+        {
+            Analyze(frame, framePointer);
+        }
+
+        protected virtual void Analyze(FileData fileData, string fileKey, string filePointer)
+        {
+        }
+
+        protected virtual void Analyze(PhysicalLocation physicalLocation, string physicalLocationPointer)
+        {
+        }
+
+        protected virtual void Analyze(Rule rule, string rulePointer)
+        {
+        }
+
+        protected virtual void Analyze(StackFrame frame, string framePointer)
+        {
+        }
+
+        protected void LogResult(ResultLevel level, string jPointer, string formatId, params string[] args)
+        {
+            Region region = GetRegionFromJPointer(jPointer);
+
+            // All messages start with "In {file}, at {jPointer}, ...". Prepend the jPointer to the args.
```

Comment:
I'm going to fix the framework so it doesn't automatically prepend the file name. I'll upgrade to the new framework, and modify my messages so they all start with "{jPointer}: ...". Then, since I _do_ in fact use the framework's logging facilities, which _do_ use the VS formatting helper, the messages will end up looking like this:

C:\Path\To\log.sarif(12,25): /runs/0/results/0/locations/0/analysisTarget/uri: Something is dreadfully wrong with the uri "foobles".

This will continue to work with the VS error list window (as it does today).

---
In reply to: [77240384](https://github.com/Microsoft/sarif-sdk/pull/501#discussion_r77240384) [](ancestors = 77240384)